src/provable/gen/transaction-bigint.ts

```diff
@@ -490,3 +511,51 @@ type AccountUpdate = {
 let AccountUpdate = provableFromLayout<AccountUpdate, Json.AccountUpdate>(
   jsLayout.AccountUpdate as any
 );
+
+type Account = {
+  publicKey: PublicKey;
+  tokenId: TokenId;
+  tokenSymbol: string;
+  balance: UInt64;
+  nonce: UInt32;
+  receiptChainHash: Field;
+  delegate?: PublicKey;
+  votingFor: Field;
+  timing: {
+    isTimed: Bool;
+    initialMinimumBalance: UInt64;
+    cliffTime: UInt32;
+    cliffAmount: UInt64;
+    vestingPeriod: UInt32;
+    vestingIncrement: UInt64;
+  };
+  permissions: {
+    editState: AuthRequired;
+    send: AuthRequired;
+    receive: AuthRequired;
+    setDelegate: AuthRequired;
+    setPermissions: AuthRequired;
+    setVerificationKey: AuthRequired;
+    setZkappUri: AuthRequired;
+    editSequenceState: AuthRequired;
+    setTokenSymbol: AuthRequired;
+    incrementNonce: AuthRequired;
+    setVotingFor: AuthRequired;
+  };
+  zkapp?: {
+    appState: Field[];
+    verificationKey?: {
```

Comment:
Just a question: can the vk even be undefined if the zkapp field exists on an account? Or maybe to phrase it different: what makes an account have a zkapp field? Is it the vk that deployed to it? just a tiny bit of curiosity :D

Reply:
yes, you can set app state even without setting the vk. I think the `zkapp` field exists as soon as one of those fields is written to, and they aren't really coupled to each other (except `provedState` which can only become true when a VK exists and appState is written to)